source/server/listener_impl.cc

```diff
@@ -512,11 +531,16 @@ void ListenerImpl::buildAccessLog() {
 }
 
 void ListenerImpl::buildInternalListener() {
-  if (config_.address().has_envoy_internal_address()) {
+  if (config_.has_internal_listener()) {
+    if (config_.has_address() || !config_.additional_addresses().empty()) {
+      throw EnvoyException(fmt::format("error adding listener '{}': address should not be used "
+                                       "when an internal listener config is provided",
+                                       name_));
```

Comment:
looks like missing a testcase for this case.

Reply:
Good catch. This is too late to check for `api_listener` so I moved it earlier.